up42/order.py

```diff
@@ -75,6 +82,39 @@ def is_fulfilled(self) -> bool:
         """
         return self.status == "FULFILLED"
 
+    @property
+    def _stac_search_order(self) -> Tuple[Client, ItemSearch]:
+        url = f"{self.auth._endpoint()}/v2/assets/stac"
+        pystac_client_aux = PySTACAuthClient(auth=self.auth).open(url=url)
+        stac_search_parameters = {
+            "max_items": MAX_ITEM,
+            "limit": LIMIT,
+            "filter": {
+                "op": "=",
+                "args": [
+                    {"property": "order_id"},
+                    self.order_id,
+                ],
+            },
+        }
+        pystac_asset_search = pystac_client_aux.search(filter=stac_search_parameters)
+        return (pystac_client_aux, pystac_asset_search)
+
+    def get_assets(self) -> List[Asset]:
+        """
+        Gets the Order assets or results.
+        """
+        if self.is_fulfilled:
+            _, pystac_asset_search = self._stac_search_order
+            assets = {
+                item.to_dict().get("properties", {}).get("up42-system:asset_id")
+                for item in pystac_asset_search.item_collection()
+            }
+            if not assets:
+                raise ValueError(f"No asset STAC metadata information available for this order {self.order_id}")
+            return [Asset(self.auth, asset_id=asset) for asset in assets]
+        raise ValueError(f"Order {self.order_id} is not FULFILLED! Status is {self.status}")
```

Comment:
nitpick: consider adding "Current":
```suggestion
        raise ValueError(f"Order {self.order_id} is not FULFILLED! Current status is {self.status}")
```

Reply:
I changed in the same commit than the other 2 but accepted :)